### ISSUE TITLE
BigQuery doesn't handle some complex STRUCT ARRAYS

### ISSUE THREAD

**Issue by tonicebrian** [sqlfluff/sqlfluff#1758]
Working with explicit struct arrays is flagged as PSR.

## Expected Behaviour
Given this valid query in BigQuery:

```
SELECT
  col_1,
  col_2
FROM
  UNNEST(ARRAY<STRUCT<col_1 STRING, col_2 STRING>>[
      ('hello','world'), 
      ('hi', 'there')
      ])
```

## Observed Behaviour
The query fails to parse

```
╰─ sqlfluff lint --dialect bigquery test.sql
== [test.sql] FAIL
L:   5 | P:  22 |  PRS | Couldn't find closing bracket for opening bracket.
All Finished 📜 🎉!
```

## Dialect
Bigquery

## Version
```
╰─ sqlfluff --version                      
sqlfluff, version 0.7.0
╰─ python --version          
Python 3.8.10
```

## Configuration
No configuration
